src/TeaPie/TeaPie.cs

```diff
@@ -116,6 +122,52 @@ public void SetVariable<T>(string name, T value, params string[] tags)
         => _variables.SetVariable(name, value, tags);
     #endregion
 
+    #region Functions
+    public FunctionsCollection CustomFunctions => _functions.CustomFunctions;
+
+    /// <summary>
+    /// Register a function with the specified <paramref name="name"/>.
+    /// </summary>
+    /// <param name="name">The name under which the function will be registered.</param>
+    /// <param name="func">Predicate of registered function.</param>
```

Comment:
Is it possible to do this using extension methods? So that the class doesn't get too big. It would only contain some internal properties for those `_functions`, and the extension class would contain the registration methods. This is how Microsoft does it in its libraries.